### DISCUSSION TITLE
How to use testAttachment or attachment within the mocha it using the last version of this plugin?

### DISCUSSION THREAD

**Discussion by Rogatto** [Shelex/cypress-allure-plugin#167] [Q&A] [answered]
I tried to use both testAttachment and attachment, but all of them are returning the following error:

 TypeError: Cannot read properties of null (reading 'addAttachment')

Because this error occurred during a `after all` hook we are skipping the remaining tests in the current suite: `Changing background color t...`
      at ./node_modules/@shelex/cypress-allure-plugin/reporter/allure-cypress/AllureInterface.js.Allure.testAttachment (webpack:///./node_modules/@shelex/cypress-allure-plugin/reporter/allure-cypress/AllureInterface.js:22:0)
      at Context.eval (webpack:///./cypress/e2e/home-page/change.background.color.spec.js:29:15)

Code:

import HomePage from "../../pages/HomePage";

const allure = Cypress.Allure.reporter.getInterface();

describe('Changing background color to white on home page', () => {

    beforeEach(() => {
        
        allure.epic('Home Page');
        allure.feature('Changing background color to white on home page');
        allure.severity('critical');
    });

    it('Clicking on the change background color bottom on home page', () => {
        cy.visit("/")
        HomePage.changeBackgroundColor()

        HomePage.elements.backGroundElement()
        .should('have.css', 'background-color')
        .and('eq', 'rgb(255, 255, 255)')

        **allure.testAttachment("test","test123","text/plain")**
        const userId = Cypress.env('userId')
        cy.log("log here: " + userId)
        cy.screenshot()
    });
});

**Comment by Shelex** [comment, marked as answer]
You are mixing up sync and async api, so at point when it is executed, async calls yet not happened.
in test I would recommend to use cy.allure().attachment()
